### PR TITLE
fix: error support

### PR DESCRIPTION
**Pull request by shanejonas**

### CODE REVIEW

File: src/RequestManager.ts [hunk 1]
```diff
@@ -49,12 +52,23 @@ class RequestManager {
   }
   private onData(data: string): void {
     const parsedData = JSON.parse(data);
-    if (typeof parsedData.result === "undefined") {
+    if (typeof parsedData.result === "undefined" && typeof parsedData.error === "undefined") {
```
Comment (zcstarr) [inline src/RequestManager.ts]: is this right do all rpc request require an error field to be defined?

File: src/RequestManager.ts [hunk 2]
```diff
@@ -49,12 +52,23 @@ class RequestManager {
...
       return;
     }
+    const req = this.requests[parsedData.id];
     // call request callback for id
-    if (this.requests[parsedData.id]) {
-      this.requests[parsedData.id](parsedData);
+    if (req) {
```
Comment (zcstarr) [inline src/RequestManager.ts]: nit. maybe unnest this request

Comment (zcstarr) [reply src/RequestManager.ts]:
```suggestion
    if(req === undefined) return;
```

File: src/RequestManager.ts (hunk 1 above)
Comment (BelfordZ) [reply src/RequestManager.ts]: as an aside...
We should get some of these standard interfaces together
```
type TJSONRPCPayload = { ... }
...
```

File: src/RequestManager.ts (hunk 2 above)
Comment (BelfordZ) [reply src/RequestManager.ts]: or throw? if this ever happens something is wrong.... no?

File: src/RequestManager.ts (hunk 1 above)
Comment (zcstarr) [reply src/RequestManager.ts]: discussed offline, its checking for the noop case.

File: src/RequestManager.ts (hunk 2 above)
Comment (zcstarr) [reply src/RequestManager.ts]: maybe, maybe not, the problem with that is 3rd party systems . I.e. my rpc service sometimes returns nothing or fires blanks

Comment (zcstarr) [reply src/RequestManager.ts]: There is a real problem of the transport failing , i.e. what happens when you socket connection gets dropped, the transport will throw , and I assume that bubbles up to the request manager. Does that result in a parsedData.error of undefined?

**Pull request merged**